gossip/src/cluster_info.rs

```diff
@@ -2729,6 +2766,34 @@ pub fn compute_retransmit_peers<T: Copy>(
     (neighbors, children)
 }
 
+pub fn compute_retransmit_peers_with_parents<'a, T: Copy>(
```

Comment:
Note this only works right now because there are only two levels of turbine. and thus each node only has two paths it can get a shred from. With any more levels, the number of paths increases by 2 for each additional turbine level, making finding the path from just the sender impossible.